### ISSUE TITLE
Consider alternate interfaces

### ISSUE THREAD

**Issue by joelnet** [acacode/kinka#1]
Having full error would be nice.

```javascript
const { err, data } = await kinka.get('/donuts')
if (err) {
  console.log('Error:' + err)
}
```

This would be consistent with node-style callbacks

```javascript
fs.readFile(file, (err, data) => {
  if (err) {
    console.log('Error:' + err)
  }
})
```

and promises:

```javascript
promisify(fs.readFile)(file)
  .then(data => {})
  .catch(err => {})
```

**Comment by js2me** [comment]
@joelnet do you think that is better to return array ?
```
const [value, success] = await kinka.get('/donuts')
```
I think that it is not a good idea because what if I want to get only properties `data` and `err` ? Using array desctructing have order of destruct

**Comment by js2me** [comment]
@joelnet I want to hear your opinion about properties of the `response` object
Currently I did properties
```
   const { status, data , err, isError, isSuccess } = await kinka.get('/donuts')
```
What do you thing about these property names?
or better rename `err` to `errorMessage` or something else

**Comment by joelnet** [comment]
I like those! I wouldn't use `isError` or `isSuccess`, I would just test the `err` object. But some people might find these useful so they could be beneficial to leave in. With destructuring, they could get what info they want.

**Comment by joelnet** [comment]
> @joelnet do you think that is better to return array ?
>
> const [value, success] = await kinka.get('/donuts')
>
> I think that it is not a good idea because what if I want to get only properties data and err ? Using array desctructing have order of destruct

Probably not. Seemed interesting at first, but it would be restrictive in which args you send back and which order you send them back in. For example, if you update your app and remove something from index 1, you would break everyones code. With object destructuring, you wouldn't have to worry about it. Ignore my initial comment about the Array.

**Comment by js2me** [comment]
@joelnet yeah! Thanks for your feedback!
Next release will have this changes

**Comment by js2me** [comment]
solved in `0.0.40--alpha`
